src/supla/sensor/sds011.h

```diff
@@ -0,0 +1,80 @@
+/*
+   Copyright (C) malarz
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation; either version 2
+   of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+   */
+
+// Dependencies:
+// https://github.com/ricki-z/SDS011
+
+#ifndef SRC_SUPLA_SENSOR_SDS011_H_
+#define SRC_SUPLA_SENSOR_SDS011_H_
+
+#include <supla/sensor/general_purpose_measurement.h>
+
+#include <SDS011.h>
+
+namespace Supla {
+namespace Sensor {
+class sds011 : public Supla::Element {
+ public:
+  // rx_pin, tx_pin: pins to which the sensor is connected
+  // refresh: time between readings (in minutes)
+  explicit sds011(int rx_pin, int tx_pin, int refresh = 10) {
+    if (refresh < 1) {
+      refresh = 10;
+    }
+    refreshIntervalMs = refresh * 60 * 1000;

```

Comment:
refreshIntervalMs is defined as uint16_t. Here you take default 10 min * 60 * 1000 and value will overflow uint16_t range.
Change it to uint32_t and add "refresh" value checking if it is valid.